### PR TITLE
SITES-10835: Documentation

### PR DESCRIPTION
**Pull request by naydav**
-- Added "Header Menu in Content Fragments Editor Extensibility" page 
-- Described multi-buttons declaration
-- Described sub items menu
-- Small fixes

### CODE REVIEW

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 1]
```diff
@@ -81,45 +84,75 @@ const guestConnection = await register({
...
-                console.log('Import button has been pressed');
-            }
         },
-    }
+    },
 });
 ```
 
-The callback is invoked when user clicks on the button and does not receive any extra parameters.
+The `onClick` callback is invoked when user clicks on the button and does not receive any extra parameters.
```
Comment (zetlen) [inline src/pages/services/aem-cf-console-admin/api/index.md:119]:
```suggestion
The `onClick` callback is invoked when a user clicks on the button. It does not receive any arguments.
```

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 2]
```diff
@@ -152,39 +190,77 @@ import { register } from "@adobe/uix-guest";
...
+|----------|-----------------------------------------------------------------------------| ------ |-------------------------------------------------------------------------------------------------------------|
+| id       | `string`                                                                    | ✔️  | **Must be unique** across all extensions. Consider adding vendor prefix to this field                       |
+| label    | `string`                                                                    | ✔️  | Button label that will be visible on UI                                                                     |
+| icon     | `string`                                                                    |    | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
+| subItems | `array`                                                                     |    | A lsit with sub menu items                                                                                  |
```
Comment (zetlen) [inline src/pages/services/aem-cf-console-admin/api/index.md:241]:
```suggestion
| subItems | `array`                                                                     |    | A list with sub menu items                                                                                  |
```

File: src/pages/services/aem-cf-console-admin/api/index.md (hunk 1 above)
Comment (zetlen) [reply src/pages/services/aem-cf-console-admin/api/index.md:119]: It's a small thing: "parameters" are what the function signature defines, "arguments" are the values received. I thought this was clearer, but it's just a suggestion.

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 3]
```diff
@@ -152,39 +190,77 @@ import { register } from "@adobe/uix-guest";
...
+| Field    | Type                         | Required | Description                                                                                                  |
+|----------|------------------------------| ------ |--------------------------------------------------------------------------------------------------------------|
+| id       | `string`                     | ✔️  | **Must be unique** across the current button sub menu                                                        |
+| label    | `string`                     | ✔️  | Button label that will be visible on UI                                                                      |
+| icon     | `string`                     |    | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
+| onClick  | `callback(selections): void` |  ✔️ | A callback for a button `onClick` event. Receives list of content fragments selected in the list             |
+
+**ContentFragment API**
+
+The structure details can be found in `@aem-sites/headless-sdk`
```
Comment (kokoc) [inline src/pages/services/aem-cf-console-admin/api/index.md:255]: `Structure details` words are confusing to me. Please explain in the docs how this related to the other parts of this docs. Also, this package seems to be private and not visible to our users. Should we have a copy of interface here?

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 4]
```diff
@@ -152,39 +190,77 @@ import { register } from "@adobe/uix-guest";
...
+**Button API**
+
+| Field    | Type                                                                        | Required | Description                                                                                                 |
+|----------|-----------------------------------------------------------------------------| ------ |-------------------------------------------------------------------------------------------------------------|
+| id       | `string`                                                                    | ✔️  | **Must be unique** across all extensions. Consider adding vendor prefix to this field                       |
+| label    | `string`                                                                    | ✔️  | Button label that will be visible on UI                                                                     |
+| icon     | `string`                                                                    |    | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
```
Comment (zetlen) [inline src/pages/services/aem-cf-console-admin/api/index.md:240]:
```suggestion
| icon     | `string`                                                                    |    | Name of a [React-Spectrum workflow icon](https://react-spectrum.adobe.com/react-spectrum/workflow-icons.html#available-icons). |
```

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 5]
```diff
@@ -152,39 +190,77 @@ import { register } from "@adobe/uix-guest";
...
+The structure details can be found in `@aem-sites/headless-sdk`
+
+```js
+import { ContentFragment } from "@aem-sites/headless-sdk";
+```
 
 ### Content Fragment Grid Columns
 
-The content fragement grid is area above below the action bar. It host the list of all the content fragments in the current view.
+The content fragment grid is area above below the action bar. It host the list of all the content fragments in the current view.
```
Comment (zetlen) [inline src/pages/services/aem-cf-console-admin/api/index.md:263]:
```suggestion
The content fragment grid is area above below the action bar. It contains the list of all the content fragments in the current view.
```

File: src/pages/services/aem-cf-console-admin/extension-development/index.md [hunk 6]
```diff
@@ -176,25 +186,27 @@ function ExtensionRegistration() {
...
-                title: 'Manage Language Copy: ' + selections[0].name,
-                url: url,
-              })
+            getButtons() {
+              return [
+                {
+                   id: "my.company.manage-language-copies",
+                   label: "Manage Language Copies",
+                   icon: 'PublishCheck', // Spectrum workflow icon code from https://spectrum.adobe.com/page/icons/
+                   onClick: (selections) => { // button's onClick callback
```
Comment (zetlen) [inline src/pages/services/aem-cf-console-admin/extension-development/index.md:198]:
```suggestion
                   onClick: (selections) => {
```

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 7]
```diff
@@ -152,39 +190,77 @@ import { register } from "@adobe/uix-guest";
...
-| Field | Type | Required | Description |
-| ----- | ---- | -------- | ----------- |
-| id | `string` | ✔️    | **Must be unique** across all extensions. Consider adding vendor prefix to this field. |
-| label | `string` | ✔️    | Button label that will be visible on UI |
-| icon | `string` |     | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
+**Button API**
+
+| Field    | Type                                                                        | Required | Description                                                                                                 |
+|----------|-----------------------------------------------------------------------------| ------ |-------------------------------------------------------------------------------------------------------------|
+| id       | `string`                                                                    | ✔️  | **Must be unique** across all extensions. Consider adding vendor prefix to this field                       |
```
Comment (zetlen) [inline src/pages/services/aem-cf-console-admin/api/index.md:238]:
```suggestion
| id       | `string`                                                                    | ✔️  | **Must be unique** across all extensions. Consider adding a vendor prefix to this field.                       |
```

File: src/pages/services/aem-cf-editor/api/header-menu/index.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+**Sub menu item API**
+
+| Field    | Type                                                                        | Required | Description                                                                                                  |
+|----------|-----------------------------------------------------------------------------| ------ |--------------------------------------------------------------------------------------------------------------|
+| id       | `string`                                                                    | ✔️  | **Must be unique** across the current button sub menu                                                        |
+| label    | `string`                                                                    | ✔️  | Button label that will be visible on UI                                                                      |
+| icon     | `string`                                                                    |    | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
```
Comment (zetlen) [inline src/pages/services/aem-cf-editor/api/header-menu/index.md:108]:
```suggestion
| icon     | `string`                                                                    |    | Name of a [React-Spectrum workflow icon](https://react-spectrum.adobe.com/react-spectrum/workflow-icons.html#available-icons). |
```

File: src/pages/services/aem-cf-editor/api/header-menu/index.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+# Extension Points API
+
+In our example, we used snippet:
+
+```js
+const contentFragment = await guestConnection.host.contentFragment.getContentFragment();`
+```
+
+[More details](../index.md) about it and the other Extension Points API.
```
Comment (zetlen) [inline src/pages/services/aem-cf-editor/api/header-menu/index.md:119]:
```suggestion
[More details](../index.md) about this and other Extension Points APIs.
```

File: src/pages/services/aem-cf-editor/api/header-menu/index.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+|----------|-----------------------------------------------------------------------------| ------ |--------------------------------------------------------------------------------------------------------------|
+| id       | `string`                                                                    | ✔️  | **Must be unique** across the current button sub menu                                                        |
+| label    | `string`                                                                    | ✔️  | Button label that will be visible on UI                                                                      |
+| icon     | `string`                                                                    |    | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
+| onClick  | `callback(): void`                                                          |  ✔️ | A callback for a button `onClick` event                                                                      |
+
+# Extension Points API
+
+In our example, we used snippet:
```
Comment (zetlen) [inline src/pages/services/aem-cf-editor/api/header-menu/index.md:113]:
```suggestion
In our example, we used this snippet:
```

File: src/pages/services/aem-cf-editor/api/index.md [hunk 11]
```diff
@@ -37,3 +37,43 @@ Available shared context data:
     }
 }
 ```
+
+### Get Content Fragment
+
+You can access data about the Content Fragment that is currently being edited by using API method in an extension:
```
Comment (zetlen) [inline src/pages/services/aem-cf-editor/api/index.md:43]:
```suggestion
You can access data about the Content Fragment that is currently being edited by using the `host.contentFragment.getContentFragment()` method in an extension:
```

File: src/pages/services/aem-cf-editor/api/index.md [hunk 12]
```diff
@@ -37,3 +37,43 @@ Available shared context data:
...
+        },
+    };
+    const guestConnection = await register(registrationConfig);
+}
+init().catch(console.error)
+```
+
+#### Result object
+
+This `contentFragment` object holds the last received state from AEM instance. It does not contain recent changes from Editor (no edits form Canvas, sidebar or changes on Variations) **until they are successfully saved** in AEM.
```
Comment (zetlen) [inline src/pages/services/aem-cf-editor/api/index.md:75]:
```suggestion
This `contentFragment` object holds the last received state from AEM instance. It does not contain recent changes from the Editor (no edits from the canvas, sidebar or changes to variations) **until they are successfully saved** in AEM.
```

**Review by kokoc — CHANGES_REQUESTED** [review]

File: src/pages/services/aem-cf-console-admin/api/index.md (hunk 1 above)
Comment (naydav) [reply src/pages/services/aem-cf-console-admin/api/index.md:119]: done

File: src/pages/services/aem-cf-console-admin/api/index.md [hunk 13]
```diff
@@ -81,45 +84,75 @@ const guestConnection = await register({
...
+**Button API**
+
+| Field    | Type                                                                        | Required | Description                                                                                               |
+|----------|-----------------------------------------------------------------------------| ------ |-----------------------------------------------------------------------------------------------------------|
+| id       | `string`                                                                    | ✔️  | **Must be unique** across all extensions. Consider adding vendor prefix to this field                     |
+| label    | `string`                                                                    | ✔️  | Button label that will be visible on UI                                                                   |
+| icon     | `string`                                                                    |    | An icon field accepts workflow icon code from @spectrum-icons library - https://spectrum.adobe.com/page/icons/ |
```
[the remark this answers is not in the document]
Comment (naydav) [reply src/pages/services/aem-cf-console-admin/api/index.md:133]: Done

File: src/pages/services/aem-cf-console-admin/api/index.md (hunk 5 above)
Comment (naydav) [reply src/pages/services/aem-cf-console-admin/api/index.md:263]: done

File: src/pages/services/aem-cf-editor/api/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (naydav) [reply src/pages/services/aem-cf-editor/api/index.md:43]: We removed "persist" method from "extensibility" API due to a few concerns:
-) bypassing all field validations;
-) exposing low level API access (just on the edge of the backend API calls) that after a release we need to support;
-) unstable the current API version (request structure, PATCH approach).

**Pull request merged**